### PR TITLE
Fix calorie commands, shorten command words, and sync docs to code

### LINKED ISSUE #222: [PE-D][Tester B] Unnecessarily complicated commands for client
Client related commands such as `add-client`, `edit-client`, `reassign-client` can be renamed to `add-c`, `edit-c`, `reassign-c` and it would not cause any confusion for the user. The app which is intended to manage potentially large number of gym users should make the commands more efficient for typing. 

The bug is considered medium because it causes occasional inconvenience to some users, but they can continue to use the product.

-------------
Labels: `severity.Medium` `type.FeatureFlaw`

### LINKED ISSUE #224: [PE-D][Tester B] Unnecessarily complicated commands for trainers
Trainer related commands such as `add-trainer`, `edit-trainer` can be renamed to `add-t`, `edit-t`, and it would not cause any confusion for the user. The app which is intended to manage potentially large number of gym users should make the commands more efficient for typing.

The bug is considered medium because it causes occasional inconvenience to some users, but they can continue to use the product.

-------------
Labels: `severity.Medium` `type.FeatureFlaw`

### LINKED ISSUE #237: [PE-D][Tester C] find error message does not identify the specific invalid character in the keyword
Steps to reproduce:
Run: `find Bob@`

Expected: Error message specifies that @ is an invalid character in a keyword, or at least highlights the character restriction clearly.

Actual: Generic invalid command format message: `Invalid command format!...`

-------------
Labels: `severity.Low` `type.FunctionalityBug`

### LINKED ISSUE #245: [PE-D][Tester C] DG states calorieTarget = 0 means "not set", but set-calorie-target 1 cal/0 is rejected
Steps to reproduce:
Run: `set-calorie-target 1 cal/0`

Expected: Per the DG, either 0 is accepted as a way to clear/unset the calorie target, or the DG should not document 0 as a valid value.

Image
<br></br>
Actual: Command fails with `Calories must be a positive integer.` — contradicting the DG's claim that 0 means "not set".

-------------
Labels: `severity.Medium` `type.FunctionalityBug`

### PR DESCRIPTION
**Pull request by TheSputnikSpacecraft**
set-cal rejects cal/0 despite DG documenting it as valid for clearing the target. log-calorie overflow error exposes Integer.MAX_VALUE and references a non-existent reset command. Trainer and client command words are unnecessarily verbose. UG documents list as clearing trainer GUI selection, but ListCommand does not call clearSelectedTrainer().

Let's,
* add parseCalorieTarget() to accept 0 as a valid calorie target value
* update SetCalorieTargetCommand to treat cal/0 as a clear operation
* fix log-calorie overflow error to a user-friendly message
* rename add-trainer, edit-trainer to add-t, edit-t
* rename add-client, edit-client, reassign-client to add-c, edit-c, reassign-c
* correct UG list command description to match actual code behavior
* update all UG and DG references to reflect renamed commands

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -334,12 +334,12 @@ To keep the text readable, the flow is split into two smaller sequence diagrams.
...
-* `edit-client INDEX [cal/CALORIES] [f/FOCUS] [r/REMARK] [v/VALIDITY]` updates multiple client-only fields in one command.
+* `edit-c INDEX [cal/CALORIES] [f/FOCUS] [r/REMARK] [v/VALIDITY]` updates multiple client-only fields in one command.
    * For `f/`, `r/`, and `v/`, providing an empty value (e.g., `f/`) clears the corresponding optional field.
 * `remark INDEX r/REMARK` updates `Client#remark`.
-* `set-calorie-target INDEX cal/CALORIES` updates `Client#calorieTarget`.
-* `log-calorie INDEX cal/CALORIES` adds to `Client#calorieIntake` rather than overwriting it.
-* `reassign-client CLIENT_INDEX t/TRAINER_INDEX` reads both the filtered client list and filtered trainer list and updates `trainerPhone` + `trainerName`.
+* `set-cal c/INDEX cal/CALORIES` updates `Client#calorieTarget`. Use `cal/0` to clear the target.
+* `log-cal c/INDEX cal/CALORIES` adds to `Client#calorieIntake` rather than overwriting it.
+* `reassign-c CLIENT_INDEX t/TRAINER_INDEX` reads both the filtered client list and filtered trainer list and updates `trainerPhone` + `trainerName`.
```
Comment (AdibShifas) [inline docs/DeveloperGuide.md:342]: Thank you for changing the documentation accordingly.

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 2]
```diff
@@ -113,6 +113,21 @@ public static int parseCalories(String calories) throws ParseException {
...
+     * Parses a {@code String calories} into a non-negative {@code int} calorie target.
+     * Accepts 0 (meaning "not set") or any positive integer.
+     *
+     * @throws ParseException if the given {@code calories} is not a non-negative integer.
+     */
+    public static int parseCalorieTarget(String calories) throws ParseException {
+        requireNonNull(calories);
+        String trimmed = calories.trim();
+        if (!trimmed.equals("0") && !StringUtil.isNonZeroUnsignedInteger(trimmed)) {
+            throw new ParseException("Calorie target must be a non-negative integer (use 0 to clear).");
```
Comment (AdibShifas) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:126]: Nice catch for this edge case!

**Review by AdibShifas — COMMENTED** [review]
LGTM!
